src/Orleans/Configuration/Options/SiloMessagingOptions.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Orleans.Runtime;
+using System;
+
+namespace Orleans.Configuration
+{
+    /// <summary>
+    /// Specifies global messaging options that are silo related.
+    /// </summary>
+    public class SiloMessagingOptions : MessagingOptions
+    {
+        /// <summary>
+        /// The SiloSenderQueues attribute specifies the number of parallel queues and attendant threads used by the silo to send outbound
+        /// messages (requests, responses, and notifications) to other silos.
+        /// If this attribute is not specified, then System.Environment.ProcessorCount is used.
+        /// </summary>
+        public int SiloSenderQueues { get; set; } = Environment.ProcessorCount;
```

Comment:
I don't know if having an environment based default is the right approach for Options. Maybe this should be constant, and use the IPostConfigure infrastructure to configure it if it remains unset.

Reply:
I realize after talking to Attila that this comment wasn't clear enough. What I meant is that the options POCO class should be very basic. Reading data from the environment is non-trivial (easy to do, but open to unforeseen side effects). For example, what if you send options to a remote server with different environment, or have a different default in a different platform.
Defaults for the strongly typed options should be constants/statics that do not change due to environment. That said, the Silo runtime itself does want to set it's own defaults based on the environment. That's where the PostConfigure method comes in, so by default the host builder can be configured to do the following to normalize the options:
```
services.PostConfigure<SiloMessagingOptions>(options => 
{
    if (options.SiloSenderQueues == 0) { options.SiloSenderQueues = Environment.ProcessorCount;
    if (options.GatewaySenderQueues == 0) { options.GatewaySenderQueues = Environment.ProcessorCount;
});